pyvista/core/utilities/arrays.py

```diff
@@ -0,0 +1,732 @@
+        PyVista or VTK dataset.
+
+    name : str | int
+        Name or index of the array.
+
+    Returns
+    -------
+    pyvista.pyvista_ndarray or None
+        Wrapped array if the index or name is valid. Otherwise, ``None``.
+
+    """
+    return _assoc_array(obj, name, 'cell')
+
+
+def row_array(obj, name):
+    """Return row array of a vtk object.
+
+    Parameters
+    ----------
+    obj : vtk.vtkDataSet
+        PyVista or VTK dataset.
+
+    name : str
+        Name of the array.
+
+    Returns
+    -------
+    numpy.ndarray
+        Wrapped array.
+
+    """
+    vtkarr = obj.GetRowData().GetAbstractArray(name)
+    return convert_array(vtkarr)
+
+
+def get_vtk_type(typ):
+    """Look up the VTK type for a given numpy data type.
+
+    Corrects for string type mapping issues.
+
+    Parameters
+    ----------
+    typ : numpy.dtype
+        Numpy data type.
+
+    Returns
+    -------
+    int
+        Integer type id specified in ``vtkType.h``
+
+    """
+    typ = _vtk.get_vtk_array_type(typ)
+    # This handles a silly string type bug
+    if typ == 3:
+        return 13
+    return typ
+
+
+def vtk_bit_array_to_char(vtkarr_bint):
+    """Cast vtk bit array to a char array.
+
+    Parameters
+    ----------
+    vtkarr_bint : vtk.vtkBitArray
+        VTK binary array.
+
+    Returns
+    -------
+    vtk.vtkCharArray
+        VTK char array.
+
+    Notes
+    -----
+    This performs a copy.
+
+    """
+    vtkarr = _vtk.vtkCharArray()
+    vtkarr.DeepCopy(vtkarr_bint)
+    return vtkarr
+
+
+def vtk_id_list_to_array(vtk_id_list):
+    """Convert a vtkIdList to a NumPy array.
+
+    Parameters
+    ----------
+    vtk_id_list : vtk.vtkIdList
+        VTK ID list.
+
+    Returns
+    -------
+    numpy.ndarray
+        Array of IDs.
+
+    """
+    return np.array([vtk_id_list.GetId(i) for i in range(vtk_id_list.GetNumberOfIds())])
+
+
+def convert_string_array(arr, name=None):
+    """Convert a numpy array of strings to a vtkStringArray or vice versa.
+
+    Parameters
+    ----------
+    arr : numpy.ndarray
+        Numpy string array to convert.
+
+    name : str, optional
+        Name to set the vtkStringArray to.
+
+    Returns
+    -------
+    vtkStringArray
+        VTK string array.
+
+    Notes
+    -----
+    Note that this is terribly inefficient. If you have ideas on how
+    to make this faster, please consider opening a pull request.
+
+    """
+    if isinstance(arr, np.ndarray):
+        # VTK default fonts only support ASCII. See https://gitlab.kitware.com/vtk/vtk/-/issues/16904
+        if np.issubdtype(arr.dtype, np.str_) and not ''.join(arr).isascii():  # avoids segfault
+            raise ValueError(
+                'String array contains non-ASCII characters that are not supported by VTK.'
+            )
+        vtkarr = _vtk.vtkStringArray()
+        ########### OPTIMIZE ###########
+        for val in arr:
+            vtkarr.InsertNextValue(val)
+        ################################
```

Comment:
What is this optimization? If it is optimization, I think a comment is in order, but want a detailed explanation.

Reply:
This is pre-existing, the blame is currently messed up. Cannot address in this PR